drivers/l3gxxxx/Kconfig

```
@@ -0,0 +1,457 @@
+# Copyright (c) 2021 Gunar Schorcht
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config HAVE_L3GD20H
+    bool
+    select MODULE_L3GXXXX if MODULE_SAUL_DEFAULT
+    help
+        Indicates that L3GD20H sensor is present.
+
+config HAVE_L3GD20
+    bool
+    select MODULE_L3GXXXX if MODULE_SAUL_DEFAULT
+    help
+        Indicates that L3GD20 sensor is present.
+
+config HAVE_L3G4200D_NG
+    bool
+    select MODULE_L3GXXXX if MODULE_SAUL_DEFAULT
+    help
+        Indicates that L3G4200D sensor is present.
+
+config HAVE_A3G4250D
+    bool
+    select MODULE_L3GXXXX if MODULE_SAUL_DEFAULT
+    help
+        Indicates that A3G4250D sensor is present.
+
+config HAVE_I3G4250D
+    bool
+    select MODULE_L3GXXXX if MODULE_SAUL_DEFAULT
+    help
+        Indicates that A3G4250D sensor is present.
+
+config HAVE_L3GXXXX_I2C
+    bool
+
+config HAVE_L3GXXXX_SPI
+    bool
+
+menuconfig MODULE_L3GXXXX
+    bool "L3GXXX 3-axis gyroscope sensor"
+    depends on TEST_KCONFIG
+    help
+        Driver for ST L3GXXXX 3-axis gyroscope sensor family.
+        The driver supports L3GD20H, L3GD20, L3G4200D, I3G4250D and A3G4250D.
+
+if MODULE_L3GXXXX
+
+menu "Sensor variants used"
+
+    config MODULE_L3GD20H
+        bool "L3GD20H"
+        default y if HAVE_L3GD20H
+        help
+            Enables the support for L3GD20H.
```

One option I see is to mimic what makefile is doing, defaulting to this one when no other is selected:
```suggestion
    config MODULE_L3GD20H
        bool "L3GD20H"
        default y if HAVE_L3GD20H
        default y if !MODULE_L3GD20 && !MODULE_L3G4200D_NG && !MODULE_A3G4250D && !MODULE_I3G4250D

        help
            Enables the support for L3GD20H.
```